packages/utils/src/lib/raf.ts

```diff
@@ -6,26 +6,101 @@ const isTest = () =>
 
 const rafQueue: Array<() => void> = []
 
+type RenderingMode = 'raf' | 'sixtyFps'
+type UpdateMode = 'onTick' | 'onPointerMove'
+type ReactUpdateMode = 'original' | 'throttled'
+
+// How do we schedule the updates? Either every 16ms or on raf
+/** @internal */
+export const renderingMode: RenderingMode = 'sixtyFps'
+// export const renderingMode: RenderingMode = 'raf'
+// How does we process the updates? Either onTick or onPointerMove
+/** @internal */
+export const updateMode: UpdateMode = 'onTick'
+// export const updateMode: UpdateMode = 'onPointerMove'
+// How do we notifiy React about updates? Either original (as soon as the change occurs) or throttled
+/** @internal */
+export const reactUpdateMode: ReactUpdateMode = 'throttled'
+// export const reactUpdateMode: ReactUpdateMode = 'original'
+
+let timesBetweenFrames: number[] = []
+let lastFrameTime = 0
+
 const tick = () => {
+	const now = Date.now()
+	const timeSinceLastTick = now - lastFrameTime
+	if (timeSinceLastTick < 1000) {
+		timesBetweenFrames.push(timeSinceLastTick)
+	}
+	if (timesBetweenFrames.length > 100) {
+		// eslint-disable-next-line no-console
+		console.log('resetting fps')
+		timesBetweenFrames = []
+	}
+	const average = timesBetweenFrames.reduce((a, b) => a + b, 0) / timesBetweenFrames.length
+	const fps = `${Math.round(1000 / average)}fps`
+	// eslint-disable-next-line no-console
+	console.log('time since last tick', timeSinceLastTick, fps)
+	lastFrameTime = now
+
 	const queue = rafQueue.splice(0, rafQueue.length)
 	for (const fn of queue) {
 		fn()
 	}
 }
 
+const fps = 60
+const targetTimePerFrame = 1000 / fps
 let frame: number | undefined
+let time = 0
+let last = 0
+let timePerFrame = targetTimePerFrame
 
-function raf() {
+function sixtyFps() {
 	if (frame) {
 		return
 	}
+	const now = Date.now()
+	const elapsed = now - last
 
+	// Variable frame rate
```

Comment:
i'd be curious about this commented out code for variable frame rate — what did you learn about this technique? helped/didn't help?

Reply:
Hard to tell still, just added this a bit before the standup. Will have to play more with it. My feeling is that it probably should help a bit. That said it might be a bit hard to hit the numbers right. Like we don't want to slow down the updates just because the last frame tool 1ms longer. Then also when do we start increasing the refresh rate? 

Think we should definitely explore it more.